erigon-lib/commitment/hex_patricia_hashed.go

```diff
@@ -2033,31 +2064,46 @@ func (hph *HexPatriciaHashed) Process(ctx context.Context, updates *Updates, log
 
 		default:
 		}
-
+		hph.metrics.Accounts.UpdatesInc(plainKey)
```

Comment:
why calls like that cannot be hidden behind function doing different things with same argument?
eg: `hph.metrics.Account(plainKey, isStorage=true)`
which will increment both account and storage counters, copy `plainKey` value inside .Accounts

And then all consecutive calls to `hph.metrics.Account` will reuse stored `hph.metrics.Accounts.currentPlainKey`, why to pass this key all the time?

Reply:
This makes very hard to do any changes to this metrics. We will have to change EVERY call instead of a couple exposed methods which controls metric access. EG: now, if tracing env is not provided, this code may just panic (not sure)

Reply:
ok maybe it will be suitable for parallel trie.. idk this API feels so wrong so i am thinking that metrics for parallel trie should not be used this way :)